src/Domain.LinnApps/Reports/GoodsInLogReportService.cs

```diff
@@ -40,26 +40,27 @@ public ResultsModel GoodsInLogReport(
             var toDateSearch = string.IsNullOrWhiteSpace(toDate) ? (DateTime?)null
                              : DateTime.Parse(toDate);
 
-            var data = this.goodsInLogRepository.FilterBy(
-                    x => (fromDateSearch == null || x.DateCreated >= fromDateSearch)
-                         && (toDateSearch == null || x.DateCreated <= toDateSearch)
-                         && (!createdBy.HasValue || x.CreatedBy == createdBy)
-                         && (string.IsNullOrEmpty(articleNumber) || x.ArticleNumber
-                                 .ToUpper().Contains(articleNumber.ToUpper().Trim()))
-                         && (!orderNumber.HasValue || x.OrderNumber == orderNumber)
-                         && (!quantity.HasValue || x.Quantity == quantity)
-                         && (!reqNumber.HasValue || x.ReqNumber == reqNumber)
-                         && (string.IsNullOrEmpty(storagePlace) || x.StoragePlace.ToUpper()
-                                 .Contains(storagePlace.ToUpper().Trim())))
-                .OrderByDescending(x => x.Id);
+            var data = await this.goodsInLogRepository.FilterByAsync(
+                           x => (fromDateSearch == null || x.DateCreated >= fromDateSearch)
+                                && (toDateSearch == null || x.DateCreated <= toDateSearch)
+                                && (!createdBy.HasValue || x.CreatedBy == createdBy)
+                                && (string.IsNullOrEmpty(articleNumber) || x.ArticleNumber.ToUpper()
+                                        .Contains(articleNumber.ToUpper().Trim()))
+                                && (!orderNumber.HasValue || x.OrderNumber == orderNumber)
+                                && (!quantity.HasValue || x.Quantity == quantity)
+                                && (!reqNumber.HasValue || x.ReqNumber == reqNumber)
+                                && (string.IsNullOrEmpty(storagePlace) || x.StoragePlace.ToUpper()
+                                        .Contains(storagePlace.ToUpper().Trim())));
+
+            var goodsInLogEntries = data.OrderByDescending(x => x.Id);
```

Comment:
not a change request on this PR, just an observation that we already made yesterday: we dont want to just get into the habit of doing OrderBys in memory - I'm about to PR a proposed solution